### PR TITLE
Fixed spec provider seed issues

### PR DESCRIPTION
**Pull request by duckyou**
* Redesign Base providers. Now for date providers we have BaseDataProvider and super class BaseProvider
* BaseSpecProvider have superclass BaseProvider
* BaseProvider contains seed and random object
screenshot_2017-12-22_14-33-07

### CODE REVIEW

**Comment by lk-geimfari** [comment]
@duckyou Is it done or work still in process?

File: mimesis/builtins/base.py [hunk 1]
```diff
@@ -1,12 +1,7 @@
...
-
-class BaseSpecProvider(object):
-    def __init__(self, seed: Optional[int] = None) -> None:
-        self.seed = seed
-        self.random = Random()
-
-        if seed is not None:
-            self.random.seed(self.seed)
+class BaseSpecProvider(BaseProvider):
+    # TODO: Docs
```
Comment (lk-geimfari) [inline mimesis/builtins/base.py]: """Base provider for all spec providers"""

**Comment by lk-geimfari** [comment]
@duckyou So, it does not work for `Generic`.

**Comment by lk-geimfari** [comment]
@duckyou This is why attributes for each superclass (i mean `BaseProvider`, `BaseDataProvider`) is a bad idea. It's hard to debug issue when we use `__init__` in each superclass. They should be mixins without own attributes.

**Comment by duckyou** [comment]
@lk-geimfari sorry for long response time. 😷 🏥 

> So, it does not work for Generic.

https://github.com/lk-geimfari/mimesis/blob/d16cebb9ea83bfa8e3e66a64ed3b32626e77c970/mimesis/providers/generic.py#L53-L61

Here's lazy loading seed issues:

`__getattr__` return provider with only locale attribute and it always return new provider

also seed not pass below to providers:
https://github.com/lk-geimfari/mimesis/blob/d16cebb9ea83bfa8e3e66a64ed3b32626e77c970/mimesis/providers/generic.py#L41-L51

#### Possible solution:

> `__getattr__` return provider with only locale attribute
> ...
> also seed not pass below to providers

just pass the `seed`, man!  :sunglasses:

> it always return new provider

save created instance of provider

#### Code:

```python
...
# TODO: Automate it. Mark `lazy` providers 
self.unit_system = UnitSystem(seed=self.seed)
self.file = File(seed=self.seed)
self.numbers = Numbers(seed=self.seed)
self.development = Development(seed=self.seed)
self.hardware = Hardware(seed=self.seed)
self.clothing_sizes = ClothingSizes(seed=self.seed)
self.internet = Internet(seed=self.seed)
self.path = Path(seed=self.seed)
self.payment = Payment(seed=self.seed)
self.games = Games(seed=self.seed)
self.cryptographic = Cryptographic(seed=self.seed)
```

```python
    def __getattr__(self, attrname: str):
        attribute = object.__getattribute__(self, '_' + attrname)
        if attribute and callable(attribute):
            self.__dict__[attrname] = attribute(
                locale=self.locale,
                seed=self.seed,
            )
            return self.__dict__[attrname]
```

##### Example:
```python
>>> from mimesis import Generic

>>> g = Generic('ru', seed=0x0123456789ABCDEF)
>>> g.address.address()
'ул. Октябрьская 714'
>>> g.address.address()
'ул. Ипатьевская 1046'
>>> g.address.address()
'Аллея Северодвинская 378'

>>> g = Generic('ru', seed=0x0123456789ABCDEF)
>>> g.address.address()
'ул. Октябрьская 714'
>>> g.address.address()
'ул. Ипатьевская 1046'
>>> g.address.address()
'Аллея Северодвинская 378'
```
```python
# pycharm python console object viewer
g = {Generic} Generic:ru:Russian
  personal 			= {Personal} Personal:ru:Russian
  address 			= {Address} Address:ru:Russian
  datetime 			= {Datetime} Datetime:ru:Russian
  business 			= {Business} Business:ru:Russian
  text 				= {Text} Text:ru:Russian
  food 				= {Food} Food:ru:Russian
  science 			= {Science} Science:ru:Russian
  code 				= {Code} Code:ru:Russian
  transport 			= {Transport} Transport:ru:Russian
  unit_system 			= {UnitSystem} UnitSystem:en:English
  file 				= {File} File:en:English
  numbers 			= {Numbers} Numbers:en:English
  development 			= {Development} Development:en:English
  hardware 			= {Hardware} Hardware:en:English
  clothing_sizes 		= {ClothingSizes} ClothingSizes:en:English
  internet 			= {Internet} Internet:en:English
  path 				= {Path} Path:en:English
  payment 			= {Payment} Payment:en:English
  games 			= {Games} Games:en:English
  cryptographic 		= {Cryptographic} Cryptographic:en:English
```

I can commit changes to this PR

**Comment by lk-geimfari** [comment]
@duckyou Great! Let me know when you commit changes. Thank you!

**Comment by duckyou** [comment]
@lk-geimfari here's!

**Comment by lk-geimfari** [comment]
@duckyou I'll merge it after reviewing on the local machine. Thanks!

**Pull request merged**